crates/ln-dlc-node/src/storage.rs

```diff
@@ -74,100 +40,82 @@ impl Default for TenTenOneInMemoryStorage {
-        if key == "manager" {
-            *self.manager.write() = Some(value);
-        } else if key.contains("monitors") {
-            self.monitors.write().insert(key.to_string(), value);
-        } else if key == "network_graph" {
-            *self.network_graph.write() = Some(value);
-        }
+impl KVStore for TenTenOneInMemoryStorage {
+    fn read(
+        &self,
+        primary_namespace: &str,
+        secondary_namespace: &str,
+        key: &str,
+    ) -> Result<Vec<u8>, Error> {
+        let value = match key {
+            "manager" => self.manager.read().clone(),
+            "network_graph" => self.network_graph.read().clone(),
+            "scorer" => self.scorer.read().clone(),
+            _ if primary_namespace == CHANNEL_MONITOR_PERSISTENCE_PRIMARY_NAMESPACE => {
+                self.monitors.read().get(key).cloned()
+            }
+            _ => None,
+        };
 
-        Ok(())
+        value.ok_or(Error::new(
+            ErrorKind::NotFound,
+            format!("{primary_namespace}/{secondary_namespace}/{key}"),
+        ))
     }
-}
 
-impl LDKStoreReader for TenTenOneInMemoryStorage {
-    fn read_network_graph(&self) -> Option<Vec<u8>> {
-        self.network_graph.read().clone()
-    }
+    fn write(
+        &self,
+        primary_namespace: &str,
+        _secondary_namespace: &str,
+        key: &str,
+        value: &[u8],
+    ) -> Result<(), Error> {
+        match key {
+            "manager" => *self.manager.write() = Some(value.to_vec()),
+            "network_graph" => *self.network_graph.write() = Some(value.to_vec()),
+            "scorer" => *self.scorer.write() = Some(value.to_vec()),
+            _ if primary_namespace == CHANNEL_MONITOR_PERSISTENCE_PRIMARY_NAMESPACE => {
+                self.monitors
+                    .write()
+                    .insert(key.to_string(), value.to_vec());
+            }
+            _ => tracing::warn!(primary_namespace, _secondary_namespace, key, "unknown key"),
+        }
 
-    fn read_manager(&self) -> Option<Vec<u8>> {
-        self.manager.read().clone()
+        Ok(())
     }
 
-    fn read_channelmonitors<ES: Deref, SP: Deref>(
+    fn remove(
         &self,
-        entropy_source: ES,
-        signer_provider: SP,
-    ) -> std::io::Result<
-        Vec<(
-            BlockHash,
-            ChannelMonitor<<SP::Target as SignerProvider>::Signer>,
-        )>,
-    >
-    where
-        ES::Target: EntropySource + Sized,
-        SP::Target: SignerProvider + Sized,
-    {
-        let mut res = Vec::new();
-        for entry in self.monitors.read().iter() {
-            if !entry.0.contains("monitors") {
-                continue;
+        primary_namespace: &str,
+        _secondary_namespace: &str,
+        key: &str,
+        _lazy: bool,
+    ) -> Result<(), Error> {
+        match key {
+            "manager" => *self.manager.write() = None,
+            "network_graph" => *self.network_graph.write() = None,
```

Comment:
🔧 Not critical at all, but we should add a handling for the scorer here as well.